### ISSUE TITLE
Moon tries to execute a directory that matches my shell name

### ISSUE THREAD

**Issue by GlacierWalrus** [moonrepo/moon#1703]
**Describe the bug**
```
[DEBUG 10:27:06.420] moon_cache::cache_engine  Deleted 0 artifacts and saved 0 bytes
▪▪▪▪ npm install
npm install --no-audit --no-fund (in workspace)
Error: process::stream::failed

  × Failed to execute ~/scripts/fish and stream output.
  ╰─▶ Permission denied (os error 13)
```
I'm running `moon check` in my fish shell.

I occasionally switch between shells because I'm somewhat whimsical. fish scripts aren't POSIX compatible so they're not source-able in bourne shells (some of my files make use of the existing shell), because of that I put fish scripts in a subdirectory in my scripts directory.

both `~/scripts` and `~/scripts/fish` are on my path

When I run `moon check`, it somehow decides it wants to execute `~/scripts/fish` and crashes. For the time being I've renamed my `fish` directory so this doesn't happen.

This may be easily fixable by checking if the inode is a file reference before attempting to use it, depending on where this is actually happening, it might even be a subissue in npm, though I've used npm for a long time and have not previously seen this issue.
 

**Steps to reproduce**
1. use fish shell
2. create a directory `~/test/fish/`
3. add `~/test/` and `~/test/fish/` to your fish path (not sure if both are necessary)
4. run moon check

If it doesn't work it might be a sub-issue of npm, or my instructions might be missing something

**Expected behavior**

**Screenshots**

**Environment**

```
    Shell: 3.6.0 - /home/linuxbrew/.linuxbrew/bin/fish
  Binaries:
    Node: 22.4.0 - ~/.asdf/installs/nodejs/22.4.0/bin/node
    Yarn: 1.22.22 - ~/.asdf/installs/nodejs/22.4.0/bin/yarn
    npm: 10.8.1 - ~/.asdf/plugins/nodejs/shims/npm
    [...]
   Monorepos:
    Yarn Workspaces: 1.22.22
```

**Additional context**

**Comment by milesj** [comment]
We actually fixed a similar issue recently in proto. We'll pull the same fix into moon.

**Comment by milesj** [comment]
Should be fixed in 1.29.4.
